### PR TITLE
add FAQ and Troubleshooting document

### PR DESCRIPTION
**Pull request by vankichi**
Signed-off-by: vankichi <kyukawa315@gmail.com>

### Description:

SSIA

### Environment:

- Go Version: 1.17.7
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 1.14.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [x] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/support/faq.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+## How to run Vald cluster?
+
+We recommend using helm for running Vald cluster.
+You can run by following steps.
+
+1. Install helm(v3~) and prepare Kubernetes cluster
+1. Configure helm charts as your demand
+1. Deploy by helm command
+
+## Is Vald Index Manager a necessary component?
```
[the remark this answers is not in the document]
Comment (vankichi) [reply docs/support/faq.md:77]:
```suggestion
## Is Vald Index Manager recommended using?
```

File: docs/support/faq.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+## What are the pluggable options?
+
+Vald has 3 pluggable options:
+
+1. Backup with the external storage for Vald Agent
+   - You can connect the external storage like S3, or GCS, or etc. to Vald Agent Sidecar component for backup.
+1. Algorithm of the core engine for Vald Agent
+   - We're going to add another algorithm in near future.
+1. Filtering with filter gateway
+   - you can filter the search results by own defined filter component by connecting to the filter gateway before returning the search result
```
Comment (kmrmt) [inline docs/support/faq.md:35]:
```suggestion
   - you can filter the search results by own defined egress filter component by connecting to the filter gateway before returning the search result
```

**Review by kmrmt — COMMENTED** [review]
When I read FAQ, I want categories or full-text search to find a question easily.
For example, my just idea is categorization by components.

File: docs/support/faq.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+## How to run Vald cluster?
+
+We recommend using helm for running Vald cluster.
+You can run by following steps.
+
+1. Install helm(v3~) and prepare Kubernetes cluster
+1. Configure helm charts as your demand
+1. Deploy by helm command
+
+## Is Vald Index Manager recommended using?
```
[the remark this answers is not in the document]
Comment (vankichi) [reply docs/support/faq.md:77]:
```suggestion
## Recommended using Vald Index Manager?
```

**Comment by vankichi** [comment]
@kmrmt 
Yes, we have to categorize but I'm not sure which approach is suitable :sweat_smile: 

Components, SItuations, or others...

File: docs/support/faq.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+   - you can convert object data to vector by own defined ingress filter component by connection to filter gateway before inserting
+
+Vald has three pluggable options:
+1. Backup with the external storage for Vald Agent
+    - You can connect the external storage like S3, GCS, etc., to the Vald Agent Sidecar component for backup.
+1. Algorithm of the core engine for Vald Agent
+    - We're going to add another algorithm soon.
+1. Filtering with the filter gateway
+    - You can filter the search results by your own defined filter component by connecting to the filter gateway before returning the search result
+    - You can convert object data to vector by own defined ingress filter component by connection to filter gateway before inserting
```
Comment (kmrmt) [inline docs/support/faq.md:45]: I think it looks almost same content.
```suggestion
Vald has three pluggable options:

1. Backup with the external storage for Vald Agent
   - You can connect the external storage like S3, or GCS, or etc. to Vald Agent Sidecar component for backup.
1. Algorithm of the core engine for Vald Agent
   - We're going to add another algorithm in near future.
1. Filtering with filter gateway
   - you can filter the search results by own defined filter component by connecting to the filter gateway before returning the search result
   - you can convert object data to vector by own defined ingress filter component by connection to filter gateway before inserting
```

**Review by kmrmt — CHANGES_REQUESTED** [review]

File: docs/support/faq.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
```
Comment (kevindiu) [inline docs/support/faq.md:46]: is it duplicated?

File: docs/support/faq.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+## Deployment
+
+### How to run the Vald cluster?
+
+We recommend using helm for running the Vald cluster.
```
Comment (kevindiu) [inline docs/support/faq.md:51]:
```suggestion
We recommend using helm to run and manage the Vald cluster.
```

File: docs/support/faq.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    - You can connect the external storage like S3, GCS, etc., to the Vald Agent Sidecar component for backup.
+1. Algorithm of the core engine for Vald Agent
+    - We're going to add another algorithm soon.
+1. Filtering with the filter gateway
+    - You can filter the search results by your own defined filter component by connecting to the filter gateway before returning the search result
+    - You can convert object data to vector by own defined ingress filter component by connection to filter gateway before inserting
+
+## Deployment
+
+### How to run the Vald cluster?
```
Comment (kevindiu) [inline docs/support/faq.md:49]:
```suggestion
### How to deploy the Vald cluster?
```

File: docs/support/faq.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+We recommend using helm for running the Vald cluster.
+You can run by following steps.
```
Comment (kevindiu) [inline docs/support/faq.md:52]:
```suggestion
You can deploy Vald cluster by following steps.
```

File: docs/support/faq.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+1. Install helm(v3~) and prepare Kubernetes cluster
+1. Configure helm charts as your demand
+1. Deploy by helm command
+
+## API
+
+### Is there any support for bulk insert?
```
Comment (kevindiu) [inline docs/support/faq.md:60]: We also provide other bulk operation (search, insert, update, etc.), I suggest to write it here.
```suggestion
### Is there any support for bulk operation?
```

File: docs/support/faq.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+
+Unfortunately, the current Vald cannot directly handle multi embedding spaces with a single Vald cluster.
+For handling the multi-embedding vectors in Vald, you have to do from 2 options.
+
+1. Deploy multiple Vald cluster
+1. Covert vector to new vector in the specific space by some methods
+
+### How to backup index data?
+
+There are three for backup:
```
Comment (kevindiu) [inline docs/support/faq.md:77]:
```suggestion
There are three backup methods supported by Vald:
```

File: docs/user-guides/troubleshooting.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+### Vald Agent NGT crashed at the insert process.
+
+Let’s check your container limit of memory at first.
+Vald Agent requires memory for keeping indexing on memory.
+
+## Search Operation
+
+### Vald returns no search result.
+
+It supposes there are two reasons.
```
Comment (kevindiu) [inline docs/user-guides/troubleshooting.md:16]:
```suggestion
There are two possible reasons.
```

File: docs/user-guides/troubleshooting.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+1. Indexing has not finished in Vald Agent
+   - Vald will search the nearest vectors of query from the indexing in Vald Agent.
+     If indexing does not complete yet, Vald Agent cancels searching.
```
Comment (kevindiu) [inline docs/user-guides/troubleshooting.md:20]:
```suggestion
     If the index operation is not completed yet, Vald Agent will cancel the search request.
```

File: docs/support/faq.md (hunk 9 above)
Comment (vankichi) [reply docs/support/faq.md:60]: @kevindiu 
umm...It is based on the user's question.

File: docs/user-guides/troubleshooting.md (hunk 12 above)
Comment (vankichi) [reply docs/user-guides/troubleshooting.md:20]: No passive sentence

File: docs/support/faq.md (hunk 10 above)
Comment (vankichi) [reply docs/support/faq.md:68]: In this term, I think `supported by Vald` is redundancy.

Comment (vankichi) [reply docs/support/faq.md:68]:
```suggestion
There are three for backup ways:
```

**Review by kmrmt — APPROVED** [review]

**Review by kevindiu — APPROVED** [review]

**Pull request merged**